src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/PostTaskForSearchCompletionControllerTest.java

```diff
@@ -98,6 +117,125 @@ public void should_return_a_200_and_retrieve_a_task_by_event_and_case_match() {
+            .body("tasks.size()", equalTo(1))
+            .body("tasks[0].task_state", equalTo("assigned"))
+            .body("tasks[0].case_id", equalTo(caseId))
+            .body("tasks[0].id", equalTo(taskId2))
+            .body("tasks[0].type", equalTo("reviewTheAppeal"))
+            .body("tasks[0].jurisdiction", equalTo("IA"))
+            .body("tasks[0].case_type_id", equalTo("Asylum"));
+
+        common.cleanUpTask(taskId1, REASON_COMPLETED);
+        common.cleanUpTask(taskId2, REASON_COMPLETED);
+    }
+
+    @Test
+    public void should_return_a_200_and_retrieve_multiple_tasks_by_event_and_case_match_and_assignee() {
+        final String assigneeId = getAssigneeId(authenticationHeaders);
+
+        // create a caseId
+        final String caseId = given.iCreateACcdCase();
+        log.info("Created caseId : {}", caseId);
+
+        // create a 3 tasks for caseId
+        sendMessage(caseId);
+        sendMessage(caseId);
+        sendMessage(caseId);
+
+        final List<CamundaTask> tasksList = iRetrieveATaskWithProcessVariableFilter("caseId", caseId, 3);
+        if (tasksList.size() != 3) {
+            fail("3 tasks should be created for case id: " + caseId);
+        }
+        // No user assigned to this task
+        final String taskId1 = tasksList.get(0).getId();
+
+        common.setupOrganisationalRoleAssignment(authenticationHeaders);
+        // assign user to taskId2
+        final String taskId2 = tasksList.get(1).getId();
+        restApiActions.post(
+            "task/{task-id}/assign",
+            taskId2,
+            new AssigneeRequest(assigneeId),
+            authenticationHeaders
+        );
+
+        // assign user to taskId3
+        final String taskId3 = tasksList.get(2).getId();
+        restApiActions.post(
+            "task/{task-id}/assign",
+            taskId3,
+            new AssigneeRequest(assigneeId),
+            authenticationHeaders
+        );
+
+        // search for completable
+        SearchEventAndCase searchEventAndCase = new SearchEventAndCase(
+            caseId, "requestRespondentEvidence", "IA", "Asylum");
+
+        Response result = restApiActions.post(
+            ENDPOINT_BEING_TESTED,
+            searchEventAndCase,
+            authenticationHeaders
+        );
+
+        result.then().assertThat()
+            .statusCode(HttpStatus.OK.value())
+            .contentType(APPLICATION_JSON_VALUE)
+            .body("tasks.size()", equalTo(2))
+            .body("tasks.id", containsInAnyOrder(new String[]{taskId2, taskId3}))
+            .body("tasks.case_id", everyItem(is(caseId)))
+            .body("tasks.jurisdiction", everyItem(is("IA")))
+            .body("tasks.case_type_id", everyItem(is("Asylum")))
+            .body("tasks.task_state", everyItem(is("assigned")));
+
+        common.cleanUpTask(taskId1, REASON_COMPLETED);
+        common.cleanUpTask(taskId2, REASON_COMPLETED);
+        common.cleanUpTask(taskId3, REASON_COMPLETED);
+    }
+
```

Comment:
do we need a test to also show completable tasks that are unassigned?

Reply:
The test already exists.